File: android/core/model/src/main/java/com/creapolis/solennix/core/model/PaymentSubmission.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PaymentSubmission(
    val id: String,
    @SerialName("event_id") val eventId: String,
    @SerialName("client_id") val clientId: String,
    @SerialName("user_id") val userId: String,
    val amount: Double,
    @SerialName("transfer_ref") val transferRef: String? = null,
    @SerialName("receipt_file_url") val receiptFileUrl: String? = null,
    val status: String, // "pending" | "approved" | "rejected"
    @SerialName("submitted_at") val submittedAt: String,
    @SerialName("reviewed_at") val reviewedAt: String? = null,
    @SerialName("rejection_reason") val rejectionReason: String? = null,
    @SerialName("linked_payment_id") val linkedPaymentId: String? = null,
    @SerialName("client_name") val clientName: String? = null,
    @SerialName("event_label") val eventLabel: String? = null
)

@Serializable
data class ReviewSubmissionRequest(
    val status: String, // "approved" | "rejected"
    @SerialName("rejection_reason") val rejectionReason: String? = null
)

File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/PaymentSubmission.swift
import Foundation

/// Represents a client's transfer payment submission from the public portal.
/// Organizers review and approve (creating a Payment row) or reject with notes.
public struct PaymentSubmission: Codable, Identifiable, Sendable, Hashable {
    public let id: String
    public let eventId: String
    public let clientId: String
    public let userId: String
    public let amount: Double
    public let transferRef: String?
    public let receiptFileUrl: String?
    public let status: PaymentSubmissionStatus
    public let submittedAt: String
    public let reviewedBy: String?
    public let reviewedAt: String?
    public let rejectionReason: String?
    public let linkedPaymentId: String?
    public let createdAt: String
    public let updatedAt: String
    public let clientName: String?
    public let eventLabel: String?
}

public enum PaymentSubmissionStatus: String, Codable, Sendable, Hashable {
    case pending
    case approved
    case rejected
}
